packages/back-end/src/services/experiments.ts

```diff
@@ -1961,21 +1961,27 @@ export function visualChangesetsHaveChanges({
   oldVisualChangeset: VisualChangesetInterface;
   newVisualChangeset: VisualChangesetInterface;
 }): boolean {
+  // If there are URL targeting differences
+  if (
+    !isEqual(oldVisualChangeset.urlPatterns, newVisualChangeset.urlPatterns)
+  ) {
+    return true;
+  }
+
   // If there are visual change differences
-  const oldVisualChanges = oldVisualChangeset.visualChanges.map(
+  const oldVisualChanges = oldVisualChangeset.visualChanges?.map(
     ({ css, js, domMutations }) => ({ css, js, domMutations })
   );
-  const newVisualChanges = newVisualChangeset.visualChanges.map(
+  const newVisualChanges = newVisualChangeset.visualChanges?.map(
     ({ css, js, domMutations }) => ({ css, js, domMutations })
   );
-  if (!isEqual(oldVisualChanges, newVisualChanges)) {
-    return true;
+
+  // If there are no visual changes in either set, there are no changes
+  if (!oldVisualChanges || !newVisualChanges) {
```

Comment:
I don't have full context here, but this doesn't sound right. If the old object is undefined and the new one isn't, wouldn't you want to return true here? Otherwise, you'll never get out of that broken state where visual changes is undefined.